src/goto-instrument/accelerate/acceleration_utils.cpp

```diff
@@ -1120,13 +1120,11 @@ bool acceleration_utilst::assign_array(
   {
     replace_expr(
       loop_counter, from_integer(0, loop_counter.type()), lower_bound);
-    simplify_expr(lower_bound, ns);
```

Comment:
Good catch - but I'd suggest to instead make those have an effect, e.g., by using `simplify` instead of `simplify_expr`

Reply:
done